src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubScaleMonitor.cs

```diff
@@ -0,0 +1,395 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Azure.EventHubs;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.EventHubs.Listeners
+{
+    internal class EventHubScaleMonitor : IScaleMonitor<EventHubTriggerMetrics>
+    {
+        private const string EventHubContainerName = "azure-webjobs-eventhub";
+        private const int PartitionLogIntervalInMinutes = 5;
+
+        private readonly string _functionId;
+        private readonly string _eventHubName;
+        private readonly string _consumerGroup;
+        private readonly EventHubsConnectionStringBuilder _connectionStringBuilder;
+        private readonly CloudBlobContainer _blobContainer;
+        private readonly ScaleMonitorDescriptor _scaleMonitorDescriptor;
+        private readonly ILogger _logger;
+
+        private EventHubClient _client;
+        private DateTime _nextPartitionLogTime;
+        private DateTime _nextPartitionWarningTime;
+
+        public EventHubScaleMonitor(
+            string functionId,
+            string eventHubName,
+            string consumerGroup,
+            string connectionString,
+            string storageConnectionString,
+            ILogger logger,
+            CloudBlobContainer blobContainer = null)
+        {
+            _functionId = functionId;
+            _eventHubName = eventHubName;
+            _consumerGroup = consumerGroup;
+            _logger = logger;
+            _scaleMonitorDescriptor = new ScaleMonitorDescriptor($"{_functionId}-EventHubTrigger-{_eventHubName}".ToLower());

```

Comment:
I think you need to add a comsumer group portion as well to the ID if consumer group is specified, since we need this to be unique, and you could have 2 different functions monitoring the same eventhub but different consumer groups? Is the consumerGroup string defaulted to some value when not specified by user, or is it null?

Reply:
It defaults to "$Default", I'll add it to the descriptor